android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FileUploadsDirCleaner.kt

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.file
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.support.base.log.logger.Logger
+import java.io.File
+import java.io.IOException
+
+/**
+ * A storage implementation for organizing temporal uploads metadata.
+ */
+object FileUploadsDirCleaner {
+
+    @VisibleForTesting
+    internal var fileNamesToBeDeleted: List<String> = emptyList()
+    private val scope = CoroutineScope(Dispatchers.IO)
+    val logger = Logger("UploadMetadataStorage")
+
+    /**
+     * Enqueue the [fileName] for future clean up.
+     */
+    fun enqueueForCleanup(fileName: String) {
+        fileNamesToBeDeleted += (fileName)
+        logger.info("File $fileName added to the upload cleaning queue.")
+    }
+
+    /**
+     * Remove all the temporary file uploads.
+     */
+    fun cleanRecentUploads(context: Context) {
```

Comment:
I think this and `cleanUploadsDirectory` could be suspend functions, removing the need for `CoroutineScope` as we are not cancelling it here. We can using withContext(Dispatchers.IO) to make sure the caller doesn't need to be explicit about the dispatcher.